### ISSUE TITLE
[enhancement] CRD support in sr-operator Helm chart

### ISSUE THREAD

**Issue by mingmxu** [StarRocks/starrocks-kubernetes-operator#448]
## Describe the current behavior

StarRocksCluster CRD is saved in a separated folder as shown in [here](https://github.com/StarRocks/starrocks-kubernetes-operator/tree/main/helm-charts/charts/kube-starrocks/charts/operator).  
In this structure, there's no upgrade support on CRD, since Helm by default ignore the CRD folder simply when the CRD is there already. As a result it can be handled in a delete-recreate pattern only. Implicitly there's always downtime of SR clusters since they are deleted when the old CRD is deleted, plus manually re-deploy the SR clusters after the new CRD is created.

## Describe the enhancement

Moving CRD into the regular [templates](https://github.com/StarRocks/starrocks-kubernetes-operator/tree/main/helm-charts/charts/kube-starrocks/charts/operator/templates) folder, or moving it to another sub-chard as dependency, CRD upgrade is treated as other resources.

## Additional context

SR operator/CRD is the mainstream pattern which has a global CRD and a single global controller for the CRD. It's the `perfectly legitimate way of packaging` as described [here](https://github.com/helm/community/blob/main/hips/hip-0011.md#no-special-crd-management).

Ps, CRD versioning is another topic in https://github.com/StarRocks/starrocks-kubernetes-operator/issues/445. 

cc @dengliu @yandongxiao @kevincai

**Comment by yandongxiao** [comment]
I think you are right. we can move the CRD yaml to templates directory.

**Comment by yandongxiao** [comment]
Currently, the Operator manages two CRDs, namely [starrocksclusters](https://github.com/StarRocks/starrocks-kubernetes-operator/blob/main/helm-charts/charts/kube-starrocks/charts/operator/crds/starrocks.com_starrocksclusters.yaml) and [starrockswarehouses](https://github.com/StarRocks/starrocks-kubernetes-operator/blob/main/helm-charts/charts/warehouse/crds/starrocks.com_starrockswarehouses.yaml). These two CRDs are placed in the `crds` directory of the Helm Chart. Due to the incomplete definition of CRD by Operator, we always receive requests from users to add support for certain fields, which often leads to changes in CRD.

We could have put the definition of CRD in the `template` directory to solve the problem of upgrading CRD. However, the current definition file of CRD is too large. Even after removing all comments, the size of this CRD file is still 320KB. It mainly includes definitions for starRocksFeSpec, starRocksBeSpec, starRocksCnSpec, and starRocksFeProxy. The underlying working principle of ArgoCD is: rendering YAML Manifest through Helm and then deploying it through `kubectl apply`. At this point there will be a `last-applied-config` annotation requirement where API Server requires that annotation size must be less than 256K or else an error will occur:

```
The CustomResourceDefinition "starrocksclusters.starrocks.com" is invalid: metadata.annotations: Too long: must have at most 262144 bytes
```

Therefore, if we want to solve this problem, we need to split up the CRD.

@dengliu

**Comment by dengliu** [comment]
what is causing the annotation tool long?
The error you're encountering—metadata.annotations: Too long: must have at most 262144 bytes—isn't directly related to the YAML file size limit for kubectl apply or the API server's request payload. Instead, it’s a specific constraint on the metadata.annotations field within a Kubernetes resource, in this case, a CustomResourceDefinition (CRD) named starrocksclusters.starrocks.com.

Kubernetes imposes a 262,144-byte (256 KiB) limit on the total size of the metadata.annotations map for any resource. This includes the combined length of all annotation keys and values (serialized as JSON). Your CRD’s annotations are exceeding this limit, which is why the API server is rejecting it.

**Comment by yandongxiao** [comment]
You are right. But the basic working principle of ArgoCD is: rendering YAML manifests through `Helm`, and then deploying it through `kubectl apply`. `kubectl apply` will add a `kubectl.kubernetes.io/last-applied-configuration` annotation, which aims to retain the content of the last CRD. And this CRD content exceeds 256KB in size.

You can verify if the resources deployed through ArgoCD have `kubectl.kubernetes.io/last-applied-configuration`.

**Comment by dengliu** [comment]
With setting [argocd.argoproj.io/sync-options](http://argocd.argoproj.io/sync-options): Replace=true on the ConfigMap you can force ArgoCD to use kubectl replace by this the [kubectl.kubernetes.io/last-applied-configuration](http://kubectl.kubernetes.io/last-applied-configuration) is no longer needed.
